lib/cc/workspace/exclusions.rb

```diff
@@ -0,0 +1,47 @@
+module CC
+  class Workspace
+    class Exclusions
+      attr_reader :exclude_paths
+
+      def initialize(exclude_paths)
+        @exclude_paths = exclude_paths.map { |p| normalize(p) }.compact
+      end
+
+      def include?(path)
+        exclude_paths.include?(path)
+      end
+
+      def apply?(path)
+        exclude_paths.any? do |exclude_path|
+          exclude_path.start_with?(path)
+        end
+      end
+
+      private
+
+      def normalize(pattern)
+        normalized = pattern.to_s.sub(%r{/\*\*(/\*)?$}, "")
+
+        if normalized.include?("*")
+          $stderr.puts(invalid_exclude_path_warning(pattern))
+        else
+          normalized
+        end
+      end
+
+      def invalid_exclude_path_warning(pattern)
+        <<-EOM
+WARNING: invalid exclude path: #{pattern.inspect}.
```

Comment:
My intent would be that the pattern is ignored and the warning is visible. That's not dot-com specific though, and assumes this `stderr` shows up on the build screen, which is may not.

But you might mean something else?